spec/gemini.spec.ts

```diff
@@ -1,31 +1,602 @@
 import os from "os";
+import OpenAI from "openai";
 import dotenv from "dotenv";
+// import { ChatCompletionCreateParamsNonStreaming } from "openai/resources/chat/completions";
 import { GoogleGenerativeAI, SchemaType } from "@google/generative-ai";
 import { flow } from "../src/flow";
+import { start } from "repl";
 
-dotenv.config();
+describe("Gemini Flow", () => {
+  let genAI: GoogleGenerativeAI;
+  let resolvedOptionsMock: jest.SpyInstance;
+  let platformSpy: jest.SpyInstance;
+  let releaseSpy: jest.SpyInstance;
+  let userInfoSpy: jest.SpyInstance;
 
-const apiKey = process.env.GEMINI_API_KEY;
+  beforeAll(() => {
+    dotenv.config();
+  });
 
-if (!apiKey) {
-  throw new Error("GEMINI_API_KEY environment variable is not set");
-}
+  beforeEach(() => {
+    const apiKey = process.env.GEMINI_API_KEY;
+    if (!apiKey) {
+      throw new Error("GEMINI_API_KEY environment variable is not set");
+    }
+    genAI = new GoogleGenerativeAI(apiKey);
+
+    resolvedOptionsMock = jest.spyOn(
+      Intl.DateTimeFormat.prototype,
+      "resolvedOptions",
+    );
+    resolvedOptionsMock.mockReturnValue({
+      locale: "en-TEST",
+      timeZone: "MockTimeZone",
+      calendar: "gregory",
+      numberingSystem: "latn",
+      timeZoneName: undefined,
+      hourCycle: undefined,
+    } as Intl.ResolvedDateTimeFormatOptions);
+
+    platformSpy = jest.spyOn(os, "platform").mockReturnValue("darwin");
+    releaseSpy = jest.spyOn(os, "release").mockReturnValue("MockedRelease");
+    userInfoSpy = jest.spyOn(os, "userInfo").mockReturnValue({
+      username: "mockUser",
+      uid: 1000,
+      gid: 1000,
+      shell: "/bin/fakeshell",
+      homedir: "/home/mockUser",
+    });
+  });
+
+  afterEach(() => {
+    // Restore original methods after test
+    resolvedOptionsMock.mockRestore();
+    platformSpy.mockRestore();
+    releaseSpy.mockRestore();
+    userInfoSpy.mockRestore();
+  });
+
+  it("should process a simple request", async () => {
+    const TestPrompt = "In which continent is Nigeria?";
+    flow.logPrompt(TestPrompt, "user-input");
+
+    const request = createRequest(TestPrompt, []);
+    flow.logRequest(request);
+
+    const generativeModel = genAI.getGenerativeModel(request);
+
+    let StartTime = new Date().toISOString();
+    const result = await generativeModel.generateContent(TestPrompt);
+    let EndTime = new Date().toISOString();
+
+    flow.logResponse({
+      ...result.response,
```

Comment:
response object should be the only parameter. The time is computed during processing.